File: src/kyu_8/compare_within_margin/c/compare_within_margin_test.c
#include <criterion/criterion.h>

int close_compare(int a, int b, int margin);

Test(ExampleTests, ShouldPassAllTheTestsProvided) {
  {
    const int a = 4;
    const int b = 5;
    const int margin = 0;
    
    const int expected = -1;
    const int received = close_compare(a, b, margin);
    
    cr_assert_eq(expected, received, "Expected: %d Received: %d", expected, received);
  }
  
  {
    const int a = 5;
    const int b = 5;
    const int margin = 0;
    
    const int expected = 0;
    const int received = close_compare(a, b, margin);
    
    cr_assert_eq(expected, received, "Expected: %d Received: %d", expected, received);
  }
  
  {
    const int a = 6;
    const int b = 5;
    const int margin = 0;
    
    const int expected = 1;
    const int received = close_compare(a, b, margin);
    
    cr_assert_eq(expected, received, "Expected: %d Received: %d", expected, received);
  }
  
  {
    const int a = 2;
    const int b = 5;
    const int margin = 3;
    
    const int expected = 0;
    const int received = close_compare(a, b, margin);
    
    cr_assert_eq(expected, received, "Expected: %d Received: %d", expected, received);
  }
  
  {
    const int a = 5;
    const int b = 5;
    const int margin = 3;
    
    const int expected = 0;
    const int received = close_compare(a, b, margin);
    
    cr_assert_eq(expected, received, "Expected: %d Received: %d", expected, received);
  }
  
  {
    const int a = 8;
    const int b = 5;
    const int margin = 3;
    
    const int expected = 0;
    const int received = close_compare(a, b, margin);
    
    cr_assert_eq(expected, received, "Expected: %d Received: %d", expected, received);
  }
  
  {
    const int a = 2;
    const int b = 5;
    const int margin = 3;
    
    const int expected = 0;
    const int received = close_compare(a, b, margin);
    
    cr_assert_eq(expected, received, "Expected: %d Received: %d", expected, received);
  }
}

File: src/kyu_8/compare_within_margin/python/compare_within_margin_test.py
import codewars_test as test
from solution import close_compare

@test.describe("Fixed Tests")
def fixed_tests():
    @test.it("No margin")
    def basic_test_cases():
        test.assert_equals(close_compare(4, 5), -1)
        test.assert_equals(close_compare(5, 5), 0)
        test.assert_equals(close_compare(6, 5), 1)

    @test.it("With margin of 3")
    def basic_test_cases():
        test.assert_equals(close_compare(2, 5, 3), 0)
        test.assert_equals(close_compare(5, 5, 3), 0)
        test.assert_equals(close_compare(8, 5, 3), 0)
        test.assert_equals(close_compare(8.1, 5, 3), 1)
        test.assert_equals(close_compare(1.99, 5, 3), -1)